src/resolve/resolved-dns-query.c

```diff
@@ -585,6 +586,17 @@ void dns_query_complete(DnsQuery *q, DnsTransactionState state) {
 
         q->state = state;
 
+        if (state == DNS_TRANSACTION_SUCCESS && set_size(q->manager->varlink_subscription) > 0) {
+                DnsQuestion *question = q->request_packet ? q->request_packet->question :
+                                                            NULL;
+                const char *query_name = question ? dns_question_first_name(question) : q->request_name;
+                if (query_name != NULL) {
```

Comment:
We allow to use implicit cast from pointer to boolean. I.e. `if (query_name)`

Reply:
Updated to use an implicit cast